daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions_unused_path_no_context.dfdl.xsd

```diff
@@ -28,20 +28,119 @@
     </xs:appinfo>
   </xs:annotation>
 
+
   <xs:element name="e1" type="xs:int" dfdl:lengthKind="explicit" dfdl:length="1" />
 
-  <xs:group name="unused_group">
+  <xs:group name="unused_group_1">
     <xs:sequence>
       <xs:element name="value1" type="xs:string" dfdl:lengthKind="explicit" dfdl:length="1" />
       <xs:sequence dfdl:terminator="{ ./value1 }" />
     </xs:sequence>
   </xs:group>
 
-  <xs:complexType name="unused_complex_type">
+  <xs:complexType name="unused_complex_type_1">
     <xs:sequence>
       <xs:element name="value2" type="xs:string" dfdl:lengthKind="explicit" dfdl:length="1" />
       <xs:sequence dfdl:terminator="{ ./value2 }"/>
     </xs:sequence>
   </xs:complexType>
 
+
+  <xs:group name="unused_group_2">
+    <xs:sequence>
+      <xs:sequence dfdl:terminator="{ ./value3 }" />
+    </xs:sequence>
+  </xs:group>
+
+  <xs:complexType name="unused_complex_type_2">
+    <xs:sequence>
+      <xs:sequence dfdl:terminator="{ ./value4 }"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:element name="unused_element_2">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:sequence dfdl:terminator="{ ./value5 }"/>
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+
+  <xs:group name="unused_group_3">
+    <xs:sequence>
+      <xs:element name="unused" type="xs:string" dfdl:terminator="{ ../value6 }" />
+    </xs:sequence>
+  </xs:group>
+
+  <xs:complexType name="unused_complex_type_3">
+    <xs:sequence>
+      <xs:element name="unused" type="xs:string" dfdl:terminator="{ ../value7 }" />
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:element name="unused_element_3">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="unused" type="xs:string" dfdl:terminator="{ ../../value8 }" />
+      </xs:sequence>
+    </xs:complexType>
+  </xs:element>
+
+
+  <xs:element name="unused_element_4">
+    <xs:complexType>
+      <xs:sequence>
+        <xs:element name="value9" type="xs:string" />
+        <xs:element name="nest1">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="nest2" type="xs:string" dfdl:terminator="{ ../../value9 }" />
```

Comment:
Is this line producing a warning? I see a `<warning>value9</warning>` above.

I don't think this should produce a warning. The ../.. walks up to nest1 then unused_element_4, but not past. 

unused_element_4 is an element. Had it been a type or group then yes, I can see a warning being ok. But it's an element. So why should this warn?

Reply:
Yeah, this was something weird I did notice. I think the issue is that unused_element_4 is a GlobalElementDecl, so there is no DPathElementCompileInfo associated with it. I believe only Root, ElementRef, and LocalElementDecl actually get DPathElementCompileInfos.

So in the context of nest2, `../../` goes to nothing (at least nothing that has a DPathElementCompileInfo. nest1 has no parent DPathElementCompileInfo and nest2 has no grandparent. Because of that, we think it does actually go up and out, even though it doesn't really.

Maybe a fix would be when we compile a GlobalElemntDecl because checkAllTopLevel is defined, we instead create a temporary Root (but one that isn't a distinguished root since we need this for the up and out check). This way there would be DPathElementCompileInfo for the unused_element_4, and there would not be a warning. Maybe there is a better approach? The feels kindof hacky just to avoid this warning, but maybe not?

Reply:
Doing some more playing around with this, a related issue is we don't even compile expression on global element decls that aren't used. For example, we could have a global element with an invalid expression like this:

```xml
  <xs:element name="unused" type="xs:string" dfdl:terminator="{ dfdl:neverCompiled() }" />
```

And no error is created. As above, I think if we created a fake root when compiling this it might solve the problem, but I'm not sure it's worth added complexity.

Reply:
Maybe we should just warn that some things (elements, types, groups) are unused and so are not compiled and leave it at that.